upgrade/upgrade-scripts/execute

```diff
@@ -612,7 +612,8 @@ fi
 if { [[ -n "$opt_f" ]] || [[ "$UPGRADE_TYPE" == "FULL" ]]; } &&
 	compare_versions "$CURRENT_VERSION" lt "18.0.0.0-0"; then
 	post_alert "reboot"
-	exec systemctl reboot || die "failed to reboot"
+	systemctl reboot || die "failed to reboot"
+	exec sleep infinity || die "failed to wait for the reboot"
```

Comment:
but then I think there is no way to avoid that, we will need https://github.com/delphix/dlpx-app-gate/pull/1692#pullrequestreview-1806912610 also for a complete fix if my race condition worry could actually happen.

Reply:
if that happens, I'd hope, the command returns a non-zero exit code signaling an "error".. and then the app-gate will detect that, and throw an exception.. e.g.
```
    try {
        # call "execute"
    } catch (NonZeroExitException e) {
      throw new VersionSwitchFailed(e, e.getMessage());
    }
```
so then, even in that case, we wouldn't call the `trySetBootFs` function, as the `VersionSwitchFailed` exception will prevent it..

but, think that through, and let me know if you agree or not..

Reply:
We'd need all versions that can upgrade to 17.0.. to similarly not call `trySetBootFs`, if `execute` returns non-zero.. which I think will be the case, but I haven't double checked that..